src/sqlfluff/core/templaters/dbt.py

```diff
@@ -104,7 +131,12 @@ def identity(x):
     @cached_property
     def dbt_selector_method(self):
         """Loads the dbt selector method."""
-        if "0.17" in self.dbt_version:
+        if self._get_project_dir() != os.getcwd():
+            self.dbt_selector_method = PathSelectorMethodWithRoot(
+                                                Path(self._get_project_dir()),
+                                                manifest = self.dbt_manifest, previous_state=None, arguments=[]
+                                            )
+        elif "0.17" in self.dbt_version:
```

Comment:
> > Hey @pvonglehn I still think this would be a helpful contribution. It would be nice to be able to have the exact same functionality of running `sqlfluff` commands in any directory like any other templater. I realise it's been a while, but would love to get this merged if you're able to get it over the line!
> 
> Hi @NiallRees - more than happy to keep going on this. Do you think I'm heading broadly in the right direction? If so next steps I guess would be to get all CI tests to pass, write a unit test for this specific functionality and then ask for review? Appreciate any guidance as I'm new to open source contributing.
> 
> With this current change the user would still need to specify the dbt project directory in the .sqlfluff file, which I think is a good start as the functionality is partially in place but incomplete as it stands. But if we wanted sqlfluff commands to work from any directory without having to specify this we would need to come up with another strategy.
> 
> ```
> [sqlfluff:templater:dbt]
> project_dir = <absolute path to dbt project directory>
> ```

Have you considered temporarily switching the directory of the running process to that of the dbt project and then manipulating any file paths that may have been provided as an argument to `sqlfluff`? Any solution which avoids taking on even more internal code dependencies on dbt would require much less maintenance going forward.